File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        if(isOddOrEven(n)){
            answer = add(n);
        }else {
            answer = square(n);
        }
        
        return answer;
    }
    
    public boolean isOddOrEven(int n){
        if(n % 2 == 1) return true;
        return false;   
    }
    
    public int add(int odd){
        if(odd < 1) return 0;
        return odd + add(odd - 2);
    }
    
    public int square(int even){
        if(even < 2 || even % 2 != 0) return 0;
        return even * even + square(even - 2);
    }
}

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.py
def solution(n):
    answer = 0
    # '==' 같은 비교연산자를 쓰지 않아도 0, 1에 따라 결과를 나눌 수 있음
    if n % 2 == 0:
        for i in range(2, n + 1, 2):
            # 'i**2' 같은 거듭제곱으로 사용 가능
            answer += i * i
    else:
        for j in range(1, n + 1, 2):
            answer += j
            
    return answer